scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/StepSimulation.java

```diff
@@ -0,0 +1,255 @@
+package gov.nasa.jpl.aerie.scheduler;
+
+import gov.nasa.jpl.aerie.merlin.driver.ActivityInstanceId;
+import gov.nasa.jpl.aerie.merlin.driver.MissionModel;
+import gov.nasa.jpl.aerie.merlin.driver.SerializedActivity;
+import gov.nasa.jpl.aerie.merlin.driver.SimulationResults;
+import gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine;
+import gov.nasa.jpl.aerie.merlin.driver.engine.TaskId;
+import gov.nasa.jpl.aerie.merlin.driver.timeline.LiveCells;
+import gov.nasa.jpl.aerie.merlin.driver.timeline.TemporalEventSource;
+import gov.nasa.jpl.aerie.merlin.protocol.driver.Scheduler;
+import gov.nasa.jpl.aerie.merlin.protocol.model.Task;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.merlin.protocol.types.TaskStatus;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+public class StepSimulation {
+
+  private Duration curTime;
+  private SimulationEngine engine;
+  private LiveCells cells;
+  private TemporalEventSource timeline;
+  private MissionModel<?> missionModel;
+
+  //used for looking up durations
+  private Map<SerializedActivity, TaskId> actToId;
+
+  //boolean stating whether the simulation results should be regenerated
+  private boolean areLastSimResultsDirty;
+
+  //simulation results so far
+  private SimulationResults lastSimResults;
+
+  //should the simulation be resetted everytime an activity is simulated
+  private boolean resetEveryTime;
+
+  //map from activity start time to serialized activity and its name
+  private LinkedHashMap<Duration, Pair<SerializedActivity, String>> activitiesInserted;
+
+  /*when we really want to reset the whole simulation*/
+  public void clearActInserted(){
+    activitiesInserted.clear();
+  }
+
+  public StepSimulation(MissionModel<?> missionModel){
+    this.missionModel = missionModel;
+    resetEveryTime = false;
+    activitiesInserted = new LinkedHashMap<>();
+  }
+
+  public void resetEveryTime(){
+    resetEveryTime= true;
```

Comment:
Consistency with the rest of the file:
```suggestion
    resetEveryTime = true;
```